sympy/liealgebras/tests/test_type_E.py

```diff
@@ -1,19 +1,74 @@
+from sympy.functions.elementary.miscellaneous import sqrt
+from sympy.core.numbers import Rational
+from sympy import Matrix
 from sympy.liealgebras.cartan_type import CartanType
-from sympy.matrices import Matrix
 
 def test_type_E():
     c = CartanType("E6")
-    m = Matrix(6, 6, [2, 0, -1, 0, 0, 0, 0, 2, 0, -1, 0, 0,
-        -1, 0, 2, -1, 0, 0, 0, -1, -1, 2, -1, 0, 0, 0, 0,
-        -1, 2, -1, 0, 0, 0, 0, -1, 2])
-    assert c.cartan_matrix() == m
-    assert c.dimension() == 8
-    assert c.simple_root(6) == [0, 0, 0, -1, 1, 0, 0, 0]
+    assert c.dimension() == 6
     assert c.roots() == 72
     assert c.basis() == 78
-    diag = " "*8 + "2\n" + " "*8 + "0\n" + " "*8 + "|\n" + " "*8 + "|\n"
+    diag = " "*8 + "6\n" + " "*8 + "0\n" + " "*8 + "|\n" + " "*8 + "|\n"
     diag += "---".join("0" for i in range(1, 6))+"\n"
-    diag += "1   " + "   ".join(str(i) for i in range(3, 7))
+    diag += "1   " + "   ".join(str(i) for i in range(2, 6))
     assert c.dynkin_diagram() == diag
-    posroots = c.positive_roots()
-    assert posroots[8] == [1, 0, 0, 0, 1, 0, 0, 0]
+
+
+def test_simpleroots():
+    c = CartanType("E6")
+    s = c.simple_roots()
+
+    assert s[0].tolist() == [[1, -1, 0, 0, 0, 0]]
+    assert s[3].tolist() == [[0, 0, 0, 1, 1, 0]]
+    assert s[4].tolist() == [[-Rational(1,2),-Rational(1,2),-Rational(1,2),-Rational(1,2),-Rational(1,2),sqrt(3)*Rational(1,2)]]
+    assert s[-1].tolist() == [[0, 0, 0, 1, -1, 0]]
+    c = CartanType("E7")
+    s = c.simple_roots()
+
+    assert s[0].tolist() == [[1, -1, 0, 0, 0, 0, 0]]
+    assert s[4].tolist() == [[0, 0, 0, 0, 1, 1, 0]]
+    assert s[5].tolist() == [[-Rational(1,2),-Rational(1,2),-Rational(1,2),-Rational(1,2),-Rational(1,2),-Rational(1,2), sqrt(2)*Rational(1,2)]]
+    assert s[-1].tolist() == [[0, 0, 0, 0, 1, -1, 0]]
+
+def test_pos_roots():
+    """The order of the roots can be finicky because E has a
+    lot of roots with the same weight level."""
+
+    c = CartanType("E6")
+    assert c.positive_roots() == [
+        Matrix([[1/2, 1/2, 1/2, 1/2, -1/2, sqrt(3)/2]]),
```

Comment:
`1/2` will be converted to a float `0.500000000000000`. It does currently compare equal to SymPy's rational `S.Half` but that may be changed in the future. Therefore it is better to use `S.Half` here. I would consider defining `h = S.Half` and then using `h` instead of `1/2` in the code.